src/transformers/models/t5gemma2/modular_t5gemma2.py

```diff
@@ -0,0 +1,1445 @@
+        vision_config = SiglipVisionConfig(**vision_config.to_dict())
+
+        encoder.is_decoder = False
+        encoder.dropout_rate = dropout_rate
+        encoder.attention_dropout = attention_dropout
+        self.encoder = encoder
+
+        decoder.is_decoder = True
+        decoder.use_cache = True
+        decoder.dropout_rate = dropout_rate
+        decoder.attention_dropout = attention_dropout
+        self.decoder = decoder
+
+        self.vision_config = vision_config
+
+        for special_token_key in ["bos_token_id", "pad_token_id", "eos_token_id"]:
+            if special_token_key not in kwargs:
+                kwargs[special_token_key] = getattr(decoder, special_token_key)
+
+        super().__init__(**kwargs)
+
+        self.mm_tokens_per_image = mm_tokens_per_image
+        self.boi_token_index = boi_token_index
+        self.eoi_token_index = eoi_token_index
+        self.image_token_index = image_token_index
+        self.initializer_range = initializer_range
+
+        self.is_encoder_decoder = is_encoder_decoder
+        self.use_cache = kwargs.get("use_cache", decoder.use_cache)
+        self.dropout_rate = dropout_rate
+        self.attention_dropout = attention_dropout
+        self.classifier_dropout_rate = classifier_dropout_rate
+
+        # Used in pipeline generation.
+        self.vocab_size = vocab_size
+
+    def __setattr__(self, key, value):
+        shared_attr_with_submodules = [
+            "output_hidden_states",
+            "output_attentions",
+            "_attn_implementation",
+            "dropout_rate",
+            "attention_dropout",
+            "vocab_size",
+        ]
+
+        if key in shared_attr_with_submodules:
+            setattr(self.encoder, key, value)
+            setattr(self.decoder, key, value)
+            if key == "_attn_implementation":
+                self.vision_config._attn_implementation = value
+        super().__setattr__(key, value)
+
+
+class T5Gemma2RMSNorm(Gemma3RMSNorm):
+    pass
+
+
+class T5Gemma2MLP(Gemma3MLP):
+    def __init__(self, config: T5Gemma2ModuleConfig):
+        super().__init__(config)
+        self.dropout = nn.Dropout(config.dropout_rate)
+
+    def forward(self, x):
+        hidden_states = self.act_fn(self.gate_proj(x)) * self.up_proj(x)
+        hidden_states = self.dropout(hidden_states)
+        down_proj = self.down_proj(hidden_states)
+        return down_proj
+
+
+class T5Gemma2RotaryEmbedding(Gemma3RotaryEmbedding):
+    def __init__(self, config: T5Gemma2ModuleConfig, device=None):
+        super().__init__(config, device)
+
+    @staticmethod
+    def compute_default_rope_parameters(
+        config: Optional[T5Gemma2ModuleConfig] = None,
+        device: Optional["torch.device"] = None,
+        seq_len: Optional[int] = None,
+        layer_type: Optional[str] = None,
+    ) -> tuple["torch.Tensor", float]:
+        return super().compute_default_rope_parameters(config, device, seq_len, layer_type)
```

Comment:
```suggestion
class T5Gemma2RotaryEmbedding(Gemma3RotaryEmbedding):
    pass
```
We do not modify anything, so we can just pass and directly inherit everything, no?

Reply:
Ah, thx for clarifying that's annoying. Let's keep it this way then